easycukes-scm/src/main/java/com/worldline/easycukes/scm/stepdefs/SCMStepdefs.java

```diff
@@ -202,5 +218,142 @@ public void commitInAnSVNRepository(String repository, String message)
                 DataInjector.injectData(message));
         log.info("Commit OK");
     }
+    
+    /**
+	 * Allows to create a branch in a given repository and execute git commit &
+	 * a git push command in the specified repository using the message provided
+	 * as a commit message
+	 * 
+	 * @param repository
+	 *            the location in which the git repository is located
+	 * @param message
+	 *            the message to be used as a commit message
+	 * @throws GitAPIException
+	 *             if something's going wrong while interacting with Git
+	 *             commands
+	 * @throws IOException
+	 *             if there's something wrong while committing the files
+	 */
+	@When("^I create a new branch in a git repository located in \"([^\"]*)\" having name \"([^\"]*)\" with the message \"([^\"]*)\"$")
+	public void createGitBarnch(String repository, String branchName,
+								String message) throws GitAPIException {
+		log.info(COMMITING_MESSAGE);
+		final File gitworkDir = new File(DataInjector.injectData(repository));
+
+		String username = "";
+		String password = null;
+		if (configuration.getValues().credentials != null) {
+			username = configuration.getValues().credentials.getLogin() != null ? configuration
+					.getValues().credentials.getLogin() : "";
+			password = configuration.getValues().credentials.getPassword() != null ? configuration
+					.getValues().credentials.getPassword() : "";
+		}
+		GitHelper.createBranch(gitworkDir, branchName, username, password,
+				DataInjector.injectData(message));
+		log.info(COMMIT_DONE_MESSAGE);
+	}
+
+
+	/**
+	 * Allows to delete a branch in a given repository and execute git commit &
+	 * a git push command in the specified repository using the message provided
+	 * as a commit message
+	 * 
+	 * @param repository
+	 *            the location in which the git repository is located
+	 * @param message
+	 *            the message to be used as a commit message
+	 * @throws GitAPIException
+	 *             if something's going wrong while interacting with Git
+	 *             commands
+	 * @throws IOException
+	 *             if there's something wrong while committing the files
+	 */
+	@When("^I delete a branch from a git repository located in \"([^\"]*)\" having name \"([^\"]*)\" with the message \"([^\"]*)\"$")

```

Comment:
I will correct this like 

"I delete this "...." branch from the git repository located at "...." with the message "...."